tests/testthat/test-client-cloud.R

```diff
@@ -303,6 +303,50 @@ test_that("Create application", {
   expect_equal(app$url, "http://fake-url.test.me/")
 })
 
+test_that("Create application with space id", {
+  mockServer <- mockServerFactory(list(
+    "^POST /outputs" = list(
+      content = function(methodAndPath, match, contentFile, ...) {
+        content <- jsonlite::fromJSON(readChar(contentFile, file.info(contentFile)$size))
+        expect_equal(content$application_type, "connect")
+        expect_equal(content$space, 333)
+        list(
+          "id" = 1,
+          "source_id" = 2,
+          "url" = "http://fake-url.test.me/",
+          "state" = "active"
+        )
+      }
+    ),
+    "^GET /applications/([0-9]+)" = list(
+      content = function(methodAndPath, match, ...) {
+        end <- attr(match, "match.length")[2] + match[2]
+        application_id <- strtoi(substr(methodAndPath, match[2], end))
+
+        list(
+          "id" = application_id,
+          "content_id" = 1
+        )
+      })
+  ))
+
+  restoreOpt <- options(rsconnect.http = mockServer$impl)
```

Comment:
You should be able to:
```r
withr::local_options(rsconnect.http = mockServer$impl)
```

That should replace the `options(); defer()` two-step.